src/main/java/uk/gov/ons/census/fwmt/csvservice/implementation/addresscheck/AddressCheckConverterService.java

```diff
@@ -77,19 +89,48 @@ public void convertToCanonical() throws GatewayException {
   }
 
   private void processObject(CsvToBean<AddressCheckListing> csvToBean) throws GatewayException {
-    // Gonna assume no one is gonna be stupid and run this without loading lookup
     for (AddressCheckListing addressCheckListing : csvToBean) {
-      if (postcodeLookupMap.containsKey(addressCheckListing.getPostcode().replaceAll("\\s+","").toUpperCase())) {
-        CreateFieldWorkerJobRequest createFieldWorkerJobRequest = createAddressCheckJob(addressCheckListing,
-            postcodeLookupMap.get(addressCheckListing.getPostcode().replaceAll("\\s+","").toUpperCase()));
-        gatewayActionAdapter.sendJobRequest(createFieldWorkerJobRequest, CANONICAL_ADDRESS_CHECK_CREATE_SENT);
-        gatewayEventManager.triggerEvent(String.valueOf(createFieldWorkerJobRequest.getCaseId()),
-            CSV_ADDRESS_CHECK_REQUEST_EXTRACTED);
-      } else {
+      if (postcodeLookupMap.containsKey(getPostcode(addressCheckListing))) {
+        if (addressCheckListingIsValid(postcodeLookupMap.get(getPostcode(addressCheckListing)))) {
+          CreateFieldWorkerJobRequest createFieldWorkerJobRequest = createAddressCheckJob(addressCheckListing,
+              postcodeLookupMap.get(getPostcode(addressCheckListing)));
+          gatewayActionAdapter.sendJobRequest(createFieldWorkerJobRequest, CANONICAL_ADDRESS_CHECK_CREATE_SENT);
+          gatewayEventManager.triggerEvent(String.valueOf(createFieldWorkerJobRequest.getCaseId()),
+              CSV_ADDRESS_CHECK_REQUEST_EXTRACTED);
+        } else {
+          RejectionReport rejectionReport = getRejectionReport(addressCheckListing.getCaseReference(),
```

Comment:
This and the next **else if** statement are very similar so extract to 
i.e. appendToRejectionReport(AddressCheckListing addressCheckListing, String message) 
Like above keep the event trigger in the if